src/ScaledPolynomialTransformFitter.cc

```diff
@@ -0,0 +1,465 @@
+    double maxMeasurementVariance = 0.0; // maximum of the per-match measurement uncertainties
+    double oldIntrinsicVariance = _intrinsicScatter*_intrinsicScatter;
+    std::size_t nGood = 0;
+    for (auto const & record : _data) {
+        if (!_keys.valid.isValid() || record.get(_keys.valid)) {
+            auto delta = record.get(_keys.output) - record.get(_keys.model);
+            directVariance += 0.5*delta.computeSquaredNorm();
+            double cxx = _keys.outputErr.getElement(record, 0, 0) - oldIntrinsicVariance;
+            double cyy = _keys.outputErr.getElement(record, 1, 1) - oldIntrinsicVariance;
+            double cxy = _keys.outputErr.getElement(record, 0, 1);
+            // square of semimajor axis of uncertainty error ellipse
+            double ca2 = 0.5*(cxx + cyy + std::sqrt(cxx*cxx + cyy*cyy + 4*cxy*cxy - 2*cxx*cyy));
+            maxMeasurementVariance = std::max(maxMeasurementVariance, ca2);
+            ++nGood;
+        }
+    }
+    directVariance /= nGood;
+
+    // Function that computes the -log likelihood of the current deltas with
+    // the variance modeled as described above.
+    auto logLikelihood = [this](double intrinsicVariance) {
+        // Uncertainties in the table right now include the old intrinsic scatter; need to
+        // subtract it off as we add the new one in.
+        double varDiff = intrinsicVariance - this->_intrinsicScatter*this->_intrinsicScatter;
+        double q = 0.0;
+        for (auto & record : this->_data) {
+            double x = record.get(this->_keys.output.getX()) - record.get(_keys.model.getX());
+            double y = record.get(this->_keys.output.getY()) - record.get(_keys.model.getY());
+            double cxx = this->_keys.outputErr.getElement(record, 0, 0) + varDiff;
+            double cyy = this->_keys.outputErr.getElement(record, 1, 1) + varDiff;
+            double cxy = this->_keys.outputErr.getElement(record, 0, 1);
+            double det = cxx*cyy - cxy*cxy;
+            q += (x*x*cyy - 2*x*y*cxy + y*y*cxx)/det + std::log(det);
+        }
+        return q;
+    };
+
+    // directVariance brackets the intrinsic variance from above, and this quantity
+    // brackets it from below:
+    double minIntrinsicVariance = std::max(0.0, directVariance - maxMeasurementVariance);
+
+    // Minimize the negative log likelihood to find the best-fit intrinsic variance.
+    static constexpr int BITS_REQUIRED = 16;  // solution good to ~1E-4
+    boost::uintmax_t maxIterations = 20;
+    auto result = boost::math::tools::brent_find_minima(
+        logLikelihood,
+        minIntrinsicVariance,
+        directVariance,
+        BITS_REQUIRED,
+        maxIterations
+    );
+    return std::sqrt(result.first);  // return RMS instead of variance
+}
+
+
+std::pair<double,std::size_t> ScaledPolynomialTransformFitter::rejectOutliers(
+    OutlierRejectionControl const & ctrl
+) {
+    if (!_keys.valid.isValid()) {
+        throw LSST_EXCEPT(
```

Comment:
`_keys.valid` is a `Key` for the field that's true if an object is outlier-rejected and false if it isn't.  But if we construct with `fromGrid`, there's no uncertainty and hence no outlier rejection, and that field never gets added to the schema.

(I've added that as a comment).